[11.x] Provide an error message for PostTooLargeException

patrickomeara opened the pull request:
The `ValidatePostSize` middleware ensures the POST is under the allowed size set in `php.ini`.

However the exception message is empty

```json
{
   "message": ""
}
```

This PR adds a message.

patrickomeara committed:
[11.x] Provide an error message for PostTooLargeException

rodrigopedra commented:
Not all the Exceptions extending `HttpException` will have a message (I am saying not all, instead of none as being too busy right now to test all of them).

For example:

```php
Route::get('test', function (Request $request) {
    // ensure result is a JSON
    $request->headers->set('Accept', 'application/json');

    return throw new NotFoundHttpException();
});
```

Will give you the same response:

```json
{
    "message": ""
}
```

Also, in case this is desirable, please use the default HTTP message for this exception: `Content Too Large` (reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413 )

patrickomeara committed:
Improve test

This test was producing a false positive where ->server('CONTENT_LENGTH') was returning an array. There is no need to mock Request here.

I've also added a pass to ensure it works

patrickomeara committed:
Assert Message

ahinkle commented on src/Illuminate/Http/Middleware/ValidatePostSize.php:
```diff
@@ -21,7 +21,7 @@ public function handle($request, Closure $next)
 
         if ($max > 0 && $request->server('CONTENT_LENGTH') > $max) {
-            throw new PostTooLargeException;
+            throw new PostTooLargeException('The POST data is too large.');
```
```suggestion
            throw new PostTooLargeException(__('Content Too Large'));
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413
